packages/core/contracts/financial-templates/contract-for-difference/ContractForDiffrenceCreator.sol

```diff
@@ -99,9 +104,14 @@ contract ContractForDifferenceCreator is Testable, Lockable {
                 finder,
                 financialProductLibrary,
                 customAncillaryData,
+                prepaidProposerReward,
                 timerAddress
             );
 
+        // Move prepaid proposer reward from the deployer to the newly deployed contract.
+        if (prepaidProposerReward > 0)
+            collateralToken.safeTransferFrom(msg.sender, address(cfd), prepaidProposerReward);
```

Comment:
nit: you might as well put this below the typecasting of the `address(cfd)` to avoid the extra casting.